lib/active_merchant/billing/gateways/braintree_blue.rb

```diff
@@ -827,15 +827,39 @@ def xid_or_ds_trans_id(three_d_secure_opts)
       end
 
       def add_stored_credential_data(parameters, credit_card_or_vault_id, options)
+        # Braintree has informed us that the stored_credential mapping may be incorrect
+        # In order to prevent possible breaking changes we will only apply the new logic if
+        # specifically requested. This will be the default behavior in a future release.
         return unless (stored_credential = options[:stored_credential])
 
-        parameters[:external_vault] = {}
-        if stored_credential[:initial_transaction]
-          parameters[:external_vault][:status] = 'will_vault'
+        add_external_vault(parameters, stored_credential)
+
+        if options[:new_stored_credential]
+          updated_stored_credentials(parameters, stored_credential)
         else
-          parameters[:external_vault][:status] = 'vaulted'
-          parameters[:external_vault][:previous_network_transaction_id] = stored_credential[:network_transaction_id]
+          classic_stored_credentials(parameters, stored_credential)
+        end
+      end
+
+      def updated_stored_credentials(parameters, stored_credential)
+        # Differences between classic and updated are
```

Comment:
Even with the comments, I think I'd suggestion something like `stored_credentials_v1` and `_v2` rather than "updated" and "classic", which are somewhat ambiguous when just reading code.

Reply:
And in any case, the `options` field name should match

Reply:
First part is done! what did you mean by the `options` field name should match?

Reply:
You got it already 👍

Reply:
oops, missed the change for the comment line this message is on.

Reply:
nice catch!